src/docfx/lib/json/JsonUtility.cs

```diff
@@ -445,35 +441,31 @@ private static void HandleError(object sender, Newtonsoft.Json.Serialization.Err
             // only log an error once
             if (args.CurrentObject == args.ErrorContext.OriginalObject)
             {
-                if (args.ErrorContext.Error is JsonReaderException || args.ErrorContext.Error is JsonSerializationException jse)
+                if (args.ErrorContext.Error is JsonReaderException || args.ErrorContext.Error is JsonSerializationException)
                 {
-                    var (source, message) = ParseException(args.ErrorContext.Error);
-                    t_status.Value.Peek().Errors.Add(Errors.ViolateSchema(source, message));
+                    var state = t_status.Value.Peek();
+                    state.Errors.Add(Errors.ViolateSchema(GetSourceInfo(state.Reader.CurrentToken), RewriteErrorMessage(args.ErrorContext.Error.Message)));
                     args.ErrorContext.Handled = true;
                 }
             }
         }
 
-        private static (SourceInfo, string message) ParseException(Exception ex)
+        private static Error ToError(JsonReaderException ex, string file = null)
```

Comment:
possible to get rid of null file here, like adding `file` to current `State`?

Reply:
`JsonUtility.Deserialize` currently does not have file path, so that one is null, we could at file path to `JsonUtility.Deserialize` in another PR.